velox/functions/sparksql/Arithmetic.h

```diff
@@ -35,6 +35,28 @@ struct RemainderFunction {
     if (UNLIKELY(n == 0)) {
       return false;
     }
+    if constexpr (std::is_floating_point_v<TInput>) {
+      handleFloatingPoint(result, a, n);
+    } else {
+      handleIntegral(result, a, n);
+    }
+    return true;
+  }
+
+ private:
+  template <typename TInput>
+  void handleFloatingPoint(TInput& result, const TInput& a, const TInput& n) {
```

Comment:
nit: const TInput& -> const TInput, as we don't need reference for numeric types.